py/dml/c_backend.py

```diff
@@ -1576,15 +1576,14 @@ def generate_initialize(device):
             # Functions called from init_object shouldn't throw any
             # exceptions. But we don't want to force them to insert try-catch
             # in the init method.
-            with InitFailure(device.site):
-                # Inline the init method
-                init = codegen_inline_byname(device, (), 'init', [], [],
-                                             device.site)
-                # Call hard_reset
-                hard_reset = codegen_call_byname(device.site, device, (),
-                                                 'hard_reset', [], [])
-
-            mkCompound(device.site, [init, hard_reset]).toc()
+            stmts = []
+            for method_name in ['init', 'hard_reset']:
+                method = device.get_component(method_name, 'method')
+                with InitFailure(method.site):
+                    stmts.append(codegen_call_byname(
+                        method.site, device, (), method_name, [], []))
+
+            mkCompound(device.site, stmts).toc()
```

Comment:
This has already been merged to main, why is it showing up here? What, do you need a rebase to get rid of this dumb diff?

Reply:
yes, forgot to rebase after that was moved away. Done now.

Reply:
it's dumb you have to do that. If there aren't any merge conflicts I feel like GH should compare against the target branch.